libs/libc/stream/lib_rawoutstream.c

```diff
@@ -46,27 +46,27 @@ static int rawoutstream_puts(FAR struct lib_outstream_s *this,
   FAR struct lib_rawoutstream_s *rthis =
                                 (FAR struct lib_rawoutstream_s *)this;
   int nwritten = 0;
-  int ret = 0;
 
-  while (nwritten != len)
+  do
     {
-      ret = _NX_WRITE(rthis->fd, (FAR const char *)buf + nwritten,
-                      len - nwritten);
-      if (ret <= 0)
+      nwritten = _NX_WRITE(rthis->fd, buf, len);
+      if (nwritten >= 0)
```

Comment:
So we account `0` as success?

Reply:
I think 0 is an invalid return value for write.

Reply:
Yes, so I'm wondering why it follows the "success" path

Reply:
but it's also not good to run the fail path because errno may not update at all. The best solution here is add DEBUGASSERT.